src/org/kie/jenkins/MavenCommand.groovy

```diff
@@ -93,39 +72,47 @@ def class MavenCommand {
         return this
     }
 
+    /**
+    * Overwrites all current settings config done.
+    */
+    MavenCommand withSettingsConfigBuilder(MavenSettingsConfigBuilder settingsConfigBuilder) {
+        this.settingsConfigBuilder = settingsConfigBuilder
+        return this
+    }
+
     /**
     * IF set, override `withSettingsXmlFile`
     **/
-    MavenCommand withSettingsXmlId(String settingsXmlId){
-        this.settingsXmlConfigFileId = settingsXmlId
+    MavenCommand withSettingsXmlId(String settingsXmlId) {
```

Comment:
The `with...` method returning `this` seems a clear Builder example, we should probably have an additional builder for MavenCommand like you did for MavenSettingsConfigBuilder.

> This is valid for the rest of `with...` methods

Reply:
@Ginxo doing so means we break a lot of pipelines using this code :S
I'd rather keep this interface on MavenCommand as it is not part of the PR.

But I could setup a MavenCommandBuilder and set as Deprecated the MavenCommand methods for now. wdyt ?

Reply:
I thought about that, and yes, you are right, we should keep MavenCommand API as it is, deprecate them, create a Builder class and move the code using the current code little by little.
I'm going to tag current repository status with `7.67.x`, this way we can use shared-library by using `@Library('jenkins-pipeline-shared-libraries@7.67.x')_`
@mbiarnes it is supposed the tag process also tag this repository as the rest of RHBA repositories

Reply:
I can handle the Kogito pipelines refactoring with MavenCommandBuilder.

So gonna create 2 issues for my side:
- Move code to MavenCommandBuilder
- Refactor Kogito pipelines to use the MavenCommandBuilder

Is that ok for you ?

Reply:
perfect!! many thanks

Reply:
created https://issues.redhat.com/browse/BXMSPROD-1701 and https://issues.redhat.com/browse/KOGITO-7192